server/controllers/manage/premises/placements/transfersController.test.ts

```diff
@@ -0,0 +1,423 @@
+        'placementEndDate-month': '5',
+        'placementEndDate-day': '14',
+      })
+      expect(mockSessionSave).toHaveBeenCalled()
+      expect(response.redirect).toHaveBeenCalledWith(managePaths.premises.placements.transfers.confirm(params))
+    })
+  })
+
+  describe('confirm', () => {
+    const sessionData = {
+      transferDate: '2025-04-28',
+      'transferDate-year': '2025',
+      'transferDate-month': '4',
+      'transferDate-day': '28',
+      destinationPremisesId: destinationAp.id,
+      destinationPremisesName: destinationAp.name,
+      placementEndDate: '2025-06-12',
+      'placementEndDate-year': '2025',
+      'placementEndDate-month': '6',
+      'placementEndDate-day': '12',
+    }
+
+    beforeEach(() => {
+      request.session.multiPageFormData = {
+        transfers: {
+          [placement.id]: { ...sessionData },
+        },
+      }
+    })
+
+    it('redirects to the new transfer page if there is no data in the session', async () => {
+      request.session.multiPageFormData = undefined
+
+      const requestHandler = transfersController.confirm()
+      await requestHandler(request, response, next)
+
+      expect(response.redirect).toHaveBeenCalledWith(managePaths.premises.placements.transfers.new(params))
+    })
+
+    it('renders the confirmation page with a summary of the emergency transfer', async () => {
+      when(validationUtils.fetchErrorsAndUserInput).calledWith(request).mockReturnValue(errorsAndUserInput)
+
+      const requestHandler = transfersController.confirm()
+      await requestHandler(request, response, next)
+
+      expect(response.render).toHaveBeenCalledWith('manage/premises/placements/transfers/confirm', {
+        backlink: managePaths.premises.placements.transfers.emergencyDetails(params),
+        pageHeading: 'Check the details of the transfer',
+        placement,
+        summaryList: transferSummaryList(sessionData),
+        errors: errorsAndUserInput.errors,
+        errorSummary: errorsAndUserInput.errorSummary,
+        ...sessionData,
+      })
+    })
+  })
+
+  describe('create', () => {
+    const sessionData = {
+      transferDate: '2025-04-28',
+      'transferDate-year': '2025',
+      'transferDate-month': '4',
+      'transferDate-day': '28',
+      destinationPremisesId: destinationAp.id,
+      destinationPremisesName: destinationAp.name,
+      placementEndDate: '2025-06-12',
+      'placementEndDate-year': '2025',
+      'placementEndDate-month': '6',
+      'placementEndDate-day': '12',
+    }
+
+    beforeEach(() => {
+      request.session.multiPageFormData = {
+        transfers: {
+          [placement.id]: { ...sessionData },
+        },
+      }
+    })
+
+    it('creates the transfer, clears the session and redirects to the placement page', async () => {
+      jest.spyOn(placementService, 'createEmergencyTransfer')
+
+      const requestHandler = transfersController.create()
+
+      await requestHandler(request, response, next)
+
```

Comment:
Suggest need to check that the session contains data and that they're valid.  If the user's been sitting on the confirm page for a long time, the session may have ended.

Reply:
Done!